server/server.js

```diff
@@ -1,20 +1,114 @@
+require("dotenv").config();
 const express = require("express");
 const cors = require("cors");
+const bcrypt = require("bcrypt");
+const { check } = require("express-validator");
+const jwt = require("jsonwebtoken");
+const authRouter = express.Router();
 
 const app = express();
 const port = 8080;
 app.use(express.json());
 app.use(cors());
+app.use("/auth", authRouter);
 
-app.post("/login", (req, res) => {
-  console.log(req.body);
-  res.send({
-    message: `Hello ${req.body.email}`,
+authRouter.use((req, res, next) => {
+  console.log("Authentication middleware for /auth");
+
+  next();
+});
+
+const users = [];
+
+authRouter.post("/signup", async (req, res) => {
+  const { email, password } = req.body;
+  const user = { email, password };
+
+  const userMatch = users.find((user) => {
+    return user.email === email;
+  });
+
+  if (userMatch) {
+    return res.status(400).json({
+      errors: [
+        {
+          msg: "This user already exists",
+        },
+      ],
+    });
+  } else {
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    users.push({
+      email: email,
+      password: hashedPassword,
+    });
+
+    const accessToken = jwt.sign(email, process.env.ACCESS_TOKEN_SECRET);
+    console.log("Successfully signed in");
+    return res.json({ accessToken: accessToken });
+  }
+});
+
+authRouter.post("/login", async (req, res) => {

```

Comment:
cel ta handler mi je mal "zmeden" na prvi pogled, ful enih `if`-ov in ne najbolj berljive kode.. pomoje lahko v naslednih PRjih nameniš čas da ga mal struktutiraš, ok?

Reply:
Bom v naslednjem PR to poštimal.